File: src/python/finished/reverse_integer.py
# https://leetcode.com/problems/reverse-integer/


class Solution:
    def reverse(self, x: int) -> int:
        is_neg = x < 0
        if is_neg:
            x = -x

        ans = 0
        while x:
            ans = ans * 10 + x % 10
            x //= 10

        if is_neg:
            ans = -ans
        return ans if ans >= -(1 << 31) and ans <= (1 << 31) - 1 else 0

File: src/rust/finished/reverse_integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let is_neg = x < 0;
        let mut x = if is_neg { -x } else { x };
        let mut ans = 0i32;
        while x != 0 {
            let Some(ok) = ans.checked_mul(10).and_then(|ans| ans.checked_add(x % 10)) else {
                return 0;
            };
            ans = ok;
            x /= 10;
        }
        if is_neg {
            -ans
        } else {
            ans
        }
    }
}
